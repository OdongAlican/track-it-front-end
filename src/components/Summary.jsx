import React from 'react'

const Summary = ( { activities }) => {
    let newArray = []
        activities.forEach(element => {
           newArray.push(parseInt(element.total)) 
        });

        let res = newArray.reduce((val, num) => val + num, 0)
        console.log(res)
    return(
        <div data-testid="appSummary" className="d-flex summary-section">
            {
                activities.map((val, index) => {
                    let percentageValue = parseInt(((parseInt(val.total) / res)) * 100)
                    if(percentageValue <= 25) {
                    return(
                        <div key = { index } className=" card-stuff pt-2 pb-2">
                            <div className="percentage-section" 
                            style={{ "border-right" : "5px solid #94E28E"}}>
                                <div>{ percentageValue }</div>
                                <div>%</div>
                            </div>
                            <div className="title-section"><span>{val.title}</span></div>
                        </div>
                    )
                    } else if ( percentageValue > 25 && percentageValue <= 50) {
                        return(
                        <div key = { index } className=" card-stuff pt-2 pb-2">
                            <div className="percentage-section" 
                            style={{ "border-right" : "5px solid #F24429",
                                    "border-bottom" : "5px solid #F24429"}}>
                                <div>{ percentageValue }</div>
                                <div>%</div>
                            </div>
                            <div className="title-section"><span>{val.title}</span></div>
                        </div>
                        )
                    } else if ( percentageValue > 50 && percentageValue <= 75) {
                        return(
                        <div key = { index } className=" card-stuff pt-2 pb-2">
                            <div className="percentage-section" 
                            style={{ "border-right" : "5px solid #94E28E",
                                    "border-bottom" : "5px solid #94E28E",
                                    "border-left" : "5px solid #94E28E"}}>
                                <div>{ percentageValue }</div>
                                <div>%</div>
                            </div>
                            <div className="title-section"><span>{val.title}</span></div>
                        </div>
                        )
                    } else {
                        return(
                            <div key = { index } className=" card-stuff pt-2 pb-2">
                            <div className="percentage-section" 
                            style={{ "border" : "5px solid #F24429"}}>
                                <div>{ percentageValue }</div>
                                <div>%</div>
                            </div>
                            <div className="title-section"><span>{val.title}</span></div>
                        </div>
                        )
                    }
                })
            }
        </div>
    )
}

export default Summary