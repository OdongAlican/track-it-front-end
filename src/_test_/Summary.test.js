import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Summary from '../components/Summary';

afterEach(cleanup);
const initialState = {
  authReducer: {
    output: 10,
    error: 'test',
  },
};

const mockStore = configureStore();
let store;

const activities = [
  { total: 12 },
];

it('renders Summary correctly', () => {
  store = mockStore(initialState);
  const { getAllByTestId } = render(
    <Router>
      <Provider store={store}><Summary activities={activities} /></Provider>
      ,
    </Router>,
  );
  const result = getAllByTestId('appSummary');
  expect(result[0]).toHaveTextContent('%');
});

it('renders Summary correctly', () => {
  store = mockStore(initialState);
  const { getAllByTestId } = render(
    <Router>
      <Provider store={store}><Summary activities={activities} /></Provider>
      ,
    </Router>,
  );
  const result = getAllByTestId('appSummary');
  expect(result[0]).not.toHaveTextContent('Some Random text');
});

it('matches Summary snapshot', () => {
  const appointment = renderer.create(
    <Router>
      <Provider store={store}><Summary activities={activities} /></Provider>
    </Router>,
  );
  const tree = appointment.toJSON();
  expect(tree).toMatchSnapshot();
});
