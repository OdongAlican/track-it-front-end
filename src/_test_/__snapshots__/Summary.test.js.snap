// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches Summary snapshot 1`] = `
<div
  className="d-flex summary-section"
  data-testid="appSummary"
>
  <div
    className=" card-stuff pt-2 pb-2"
  >
    <div
      className="percentage-section"
      style={
        Object {
          "border": "5px solid #F24429",
        }
      }
    >
      <div>
        100
      </div>
      <div>
        %
      </div>
    </div>
    <div
      className="title-section"
    >
      <span />
    </div>
  </div>
</div>
`;
